qiskit/dagcircuit/dagcircuit.py

```diff
@@ -122,7 +147,11 @@ def __init__(self):
     @property
     def wires(self):
         """Return a list of the wires in order."""
-        return self.qubits + self.clbits
+        return (
+            self.qubits
+            + self.clbits
+            + [var for vars in self._vars_by_type.values() for var in vars]
```

Comment:
Honestly, I was kind of giving up on the runtime-performance of this particular property for this PR because if it would be a perf problem, it already _is_ because of the recalculation.

`list(self.input_map)` would be better, I was mostly concerned about it being a different iteration order to the existing form and that having knock-on effects on some of our tracked metrics.  I can swap it over to that and we can see, if you prefer?

Reply:
It's fine changing the iteration order is a valid reason to leave this. If it becomes an issue we can revisit this in the future. Looking through the code the only thing that uses this is `CommutationAnalysis` and `CommutativeCancellation` to get an iterator over bits (I think it only cares about qubits I'll look at a fix PR for that).